src/utils/errors/error.ts

```diff
@@ -0,0 +1,54 @@
+import { CustomErrorMapping, ErrorDefinition } from './types';
+import { STRING_PREFIX_AFTER_UNICODE_REPLACEMENT, RAILGUN_ERRORS, CUSTOM_ERRORS, INVALID_ASCII_REGEX } from './constants';
+import { isDefined } from '../util';
+
+const validAscii = (str: string) => {
```

Comment:
Now that we are refactoring, perhaps we should give this a more fitting name, perhaps `sanitizeAscii`